x-pack/plugins/triggers_actions_ui/public/application/sections/rule_details/components/rule_definition.tsx

```diff
@@ -82,6 +83,17 @@ export const RuleDefinition: React.FunctionComponent<RuleDefinitionProps> = ({
         : false)
     );
   }, [hideEditButton, canSaveRule, ruleTypeRegistry, rule]);
+
+  const ruleDescription = useMemo(() => {
+    if (ruleTypeRegistry.has(rule.ruleTypeId)) {
+      return ruleTypeRegistry.get(rule.ruleTypeId).description;
+    }
+    if (rule.consumer === AlertConsumers.SIEM) {
+      return rule.params.description as string;
+    }
+    return '';
```

Comment:
it seems like security rules have their descriptions in the `params`. Not sure if there's a better fallback we can use

Reply:
We should make every effort to avoid special casing rule types - especially with regards to the content of `params` as these can and will change at which point this UI will break.

@elastic/response-ops-execution - does anyone have historical context into why Security rules are not in the ruleTypeRegistry? I assume this is because historically they were not displayed in Stack Management, but as this no longer seems to be the case, shouldn't they be in the registry like everyone else?

Reply:
oh, I just realised - the `description` in the `params` is the description of the *rule* not the *rule type*, so we're mixing domains here anyway, and these values are not actually the same.....

Reply:
>does anyone have historical context into why Security rules are not in the ruleTypeRegistry? I assume this is because historically they were not displayed in Stack Management, but as this no longer seems to be the case, shouldn't they be in the registry like everyone else?

The UI registry got created after security solution rule types were created. It is best practice to register the rule type in the UI so a description, link to the docs, etc can be added. It may be best to go that approach and register their type for them?

Reply:
Either way we shouldn't user `params` here as its for the rule type, not the rule.
If we do want to get a quick fix out then I'm ok with a hard coded description that we then replace with a proper usage of the RuleType Registry for 8.5

Reply:
@elastic/security-solution: would love to hear your thoughts on this, thanks!

Reply:
Hey @JiaweiWu @gmmorris @mikecote, I'm not familiar with the rule type registry and what is it used for. As a user of the alerting plugin, I'd expect (well, ideally) that registering rule types on the BE via `plugins.alerting.registerType` should be enough, but looks like it's not the case for some reason. Could you please open a ticket with a bit more context and your expectation from us?

Reply:
Hey @banderror we're working on improving our docs and coming up with best practices, once we have those we'll have something clear on the expectation for each rule type built. For now, more info on this UI registry can be found here: https://github.com/elastic/kibana/tree/main/x-pack/plugins/triggers_actions_ui. It encapsulates things that we cannot make the backend manage, for example, react components for the UI framework to render. Until we have the best practices documented, feel free to reach out to me if you want to discuss or learn more 👍

Reply:
> It encapsulates things that we cannot make the backend manage, for example, react components for the UI framework to render.

Ahhh right, that makes sense!

> Until we have the best practices documented, feel free to reach out to me if you want to discuss or learn more 👍

Thank you Mike. Yea, I'm interested in learning more about this. I'm going to read the docs you shared in the next few weeks, and then maybe before the start of the 8.6 cycle we could discuss it.

If you need us to do anything urgently as part of https://github.com/elastic/kibana/issues/139768, please ping me on GH or drop a message in our slack channel. Thanks @JiaweiWu and @mikecote!